pkg/deploy/kapp.go

```diff
@@ -38,15 +41,20 @@ var _ Deploy = &Kapp{}
 func NewKapp(appSuffix string, opts v1alpha1.AppDeployKapp, genericOpts ProcessedGenericOpts,
 	globalDeployRawOpts []string, cancelCh chan struct{}, cmdRunner exec.CmdRunner) *Kapp {
 
-	return &Kapp{appSuffix, opts, genericOpts, globalDeployRawOpts, cancelCh, cmdRunner}
+	return &Kapp{appSuffix, opts, genericOpts, globalDeployRawOpts, cancelCh, cmdRunner, nil}
 }
 
 // Deploy takes the output from templating, and the app name,
 // it shells out, running kapp deploy ...
 func (a *Kapp) Deploy(tplOutput string, startedApplyingFunc func(),
 	changedFunc func(exec.CmdRunResult)) exec.CmdRunResult {
 
-	args, err := a.addDeployArgs([]string{"deploy", "--prev-app", a.oldManagedName(), "-f", "-"})
+	metadataFile := fmt.Sprintf("/etc/kappctrl-mem-tmp/metadata-%s", a.genericOpts.Name)
```

Comment:
/etc/kappctrl-mem-tmp is more of an implementation detail of memdir.TmpDir. lets use that class to create a directory. this will also fix a bug above where we only use name of an app vs name+namespace.

Reply:
great catch! updated to use tmpdir.

if the tmp dir fails to create however, it will cause the deploy to fail.

Reply:
i think that is a good behaviour.

Reply:
Yeah, it will fail with an error message closer to the root cause (rather than the kapp deploy to fail and use that error message instead back to the operator)